integration/network/bridge/bridge_linux_test.go

```diff
@@ -754,3 +755,57 @@ func TestRemoveLegacyLink(t *testing.T) {
 	res = ctr.ExecT(ctx, t, c, clientId, []string{"wget", "-T3", "http://" + svrAddr})
 	assert.Check(t, is.Contains(res.Stderr(), "download timed out"))
 }
+
+// TestPortMappingRestore check that port mappings are restored when a container
+// is restarted after a daemon restart.
+//
+// Replacement for integration-cli test DockerDaemonSuite/TestDaemonIptablesCreate
+func TestPortMappingRestore(t *testing.T) {
+	skip.If(t, testEnv.IsRootless(), "fails before and after restart")
+
+	ctx := setupTest(t)
+	d := daemon.New(t)
+	d.StartWithBusybox(ctx, t)
+	defer d.Stop(t)
+	c := d.NewClientT(t)
+
+	const svrName = "svr"
+	cid := ctr.Run(ctx, t, c,
+		ctr.WithExposedPorts("80/tcp"),
+		ctr.WithPortMap(nat.PortMap{"80/tcp": {}}),
+		ctr.WithName(svrName),
+		ctr.WithRestartPolicy(containertypes.RestartPolicyUnlessStopped),
+		ctr.WithCmd("httpd", "-f"),
+	)
+	defer func() { ctr.Remove(ctx, t, c, cid, containertypes.RemoveOptions{Force: true}) }()
+
+	check := func() {
+		t.Helper()
+		insp := ctr.Inspect(ctx, t, c, cid)
+		assert.Check(t, is.Equal(insp.State.Running, true))
+		assert.Assert(t, is.Contains(insp.NetworkSettings.Ports, nat.Port("80/tcp")))
+		assert.Assert(t, is.Len(insp.NetworkSettings.Ports["80/tcp"], 2))
+		hostPort := insp.NetworkSettings.Ports["80/tcp"][0].HostPort
+		res := ctr.RunAttach(ctx, t, c,
+			ctr.WithExtraHost("thehost:host-gateway"),
+			ctr.WithCmd("wget", "-T3", "http://"+net.JoinHostPort("thehost", hostPort)),
+		)
+		// 404 means the http request worked, but the http server had nothing to serve.
+		assert.Check(t, is.Contains(res.Stderr.String(), "404 Not Found"))
```

Comment:
https://go.dev/wiki/TestComments#keep-going
```suggestion
		if assert.Check(t, is.Contains(insp.NetworkSettings.Ports, nat.Port("80/tcp"))) &&
			assert.Check(t, is.Len(insp.NetworkSettings.Ports["80/tcp"], 2)) {
			hostPort := insp.NetworkSettings.Ports["80/tcp"][0].HostPort
			res := ctr.RunAttach(ctx, t, c,
				ctr.WithExtraHost("thehost:host-gateway"),
				ctr.WithCmd("wget", "-T3", "http://"+net.JoinHostPort("thehost", hostPort)),
			)
			// 404 means the http request worked, but the http server had nothing to serve.
			assert.Check(t, is.Contains(res.Stderr.String(), "404 Not Found"))
		}
```

Reply:
Done.